shell/models/__tests__/catalog.cattle.io.app.test.ts

```diff
@@ -0,0 +1,80 @@
+import CatalogApp from '@shell/models/catalog.cattle.io.app';
+
+describe('class CatalogApp', () => {
+  describe('should return upgradeAvailable for this chart', () => {
+    describe('as false to be excluded from some logic if', () => {
+      const expectation = false;
+
+      it('is fleet', () => {
+        const catalogApp = new CatalogApp({ spec: { chart: { metadata: { annotations: { 'fleet.cattle.io/bundle-id': 'anything' } } } } });
+
+        expect(catalogApp.upgradeAvailable).toStrictEqual(expectation);
+      });
+
+      it('is managed,', () => {
+        const catalogApp = new CatalogApp({ spec: { chart: { metadata: { annotations: { 'catalog.cattle.io/managed': 'anything' } } } } });
+
+        expect(catalogApp.upgradeAvailable).toStrictEqual(expectation);
+      });
+    });
+
+    describe('as latest version', () => {
+      it('excluding pre-releases', () => {
+        const version = '1';
+        const catalogApp = new CatalogApp({});
```

Comment:
comment only. the first arg to models is the json for the object (normally received by http request). to use less lines the spec can be passed there. to prefer clarity would vote leaving as is